subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/AsdfInstallationSupplier.java

```diff
@@ -27,17 +27,25 @@
 
 public class AsdfInstallationSupplier extends AutoDetectingInstallationSupplier {
 
+    private final Provider<String> asdfDataDir;
+    private final Provider<String> asdfUserHome;
     private final Provider<String> asdfHome;
 
     @Inject
     public AsdfInstallationSupplier(ProviderFactory factory) {
         super(factory);
+        asdfDataDir = getEnvironmentProperty("ASDF_DATA_DIR");
+        asdfUserHome = getSystemProperty("user.home")
+            .map(home -> new File(home, ".asdf").getAbsolutePath());
         asdfHome = getEnvironmentProperty("ASDF_DIR");
```

Comment:
Checking in `ASDF_DIR` is in fact incorrect.

Should we remove that entirely?

Reply:
Correct. I only saw the `ASDF_DIR` environment variable and falsy assumed that is matching the data dir. Thanks for the PR, I can take it from here and get this into RC2